packages/typir/src/initialization/type-reference.ts

```diff
@@ -0,0 +1,223 @@
+/******************************************************************************
+ * Copyright 2024 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { TypeInferenceCollectorListener, TypeInferenceRule } from '../features/inference.js';
```

Comment:
I like this initialization split. This is a rather difficult topic in any case, it is easier to have an overview now.